cpp/src/binary/binary_ops.cu

```diff
@@ -510,17 +510,33 @@ DEF_BITWISE_IMPL_GROUP(xor, DeviceBitwiseXor)
 gdf_column gdf_validity_column(const gdf_column &col) {
     gdf_column ret;
     ret.data = col.valid;
+    // TODO: this will need to be changed when gdf_valid_type is changed to 4 byte
     ret.dtype = GDF_INT8;
     ret.valid = nullptr;
     ret.size = (col.size + GDF_VALID_BITSIZE - 1) / GDF_VALID_BITSIZE;
     return ret;
 }
 
 gdf_error gdf_validity_and(gdf_column *lhs, gdf_column *rhs, gdf_column *output) {
-    GDF_REQUIRE ( lhs->valid && rhs->valid && output->valid, GDF_VALIDITY_MISSING);
+    GDF_REQUIRE ( output->valid, GDF_VALIDITY_MISSING);
     gdf_column x = gdf_validity_column(*lhs);
     gdf_column y = gdf_validity_column(*rhs);
     gdf_column z = gdf_validity_column(*output);
+    
+    // TODO: this will need to be changed when gdf_valid_type is changed to 4 byte
+    if ( x.data == nullptr && y.data != nullptr ) {
+        CUDA_TRY( cudaMemcpy(z.data, y.data, y.size, cudaMemcpyDeviceToDevice) );
```

Comment:
You'll also need to update the output column's null count.

Reply:
That's not what the name of the function says. I'll add another function that does the count update in a separate PR where I'm fixing the fact that it takes two api calls to do a binary op and null mask calculation.

Reply:
Sounds good, can be addressed in another PR.